pkg/csi/service/vanilla/controller.go

```diff
@@ -2113,17 +2106,10 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 				}
 			}
 			var nodevm *cnsvsphere.VirtualMachine
-			if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.UseCSINodeId) {
-				// if node is not yet updated to run the release of the driver publishing Node VM UUID as Node ID
-				// look up Node by name
-				nodevm, err = c.nodeMgr.GetNodeByName(ctx, req.NodeId)
-				if err == node.ErrNodeNotFound {
-					log.Infof("Performing node VM lookup using node VM UUID: %q", req.NodeId)
-					nodevm, err = c.nodeMgr.GetNodeByUuid(ctx, req.NodeId)
-				}
-
-			} else {
-				nodevm, err = c.nodeMgr.GetNodeByName(ctx, req.NodeId)
+			nodevm, err = c.nodeMgr.GetNodeByName(ctx, req.NodeId)
```

Comment:
Can you clarify if `req.NodeID` is really nodeName here or `node.Spec.ProviderID` (or vm's UUID determine from node). 

For clusters which were not using the feature gate before and now are being upgraded, are the `req.NodeID` going to change from `nodeName` to `node.Spec.ProviderID` ?

Reply:
req.NodeID will be the node ID we have published on the CSINodes object.
To consider upgrade impact, we need to ensure we first look up node VM considering req.NodeID as Node name, if it fails then we do need to look up by considering req.nodeID as VM UUID.

eventually, the goal is to remove dependency on `node.Spec.ProviderID`. Look-up will not be done using `node.Spec.ProviderID` after this change.